test/rx_ext/serial_disposable_test.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:flutter_reactive_ble/flutter_reactive_ble.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+
+abstract class _Disposer {
+  Future<void> dispose(int value);
+}
+
+class _DisposerMock extends Mock implements _Disposer {}
```

Comment:
Mocks are not the goal, tests are the goal. Put what matters higher in the file, so a reader doesn't have to go through the unimportant stuff first.

Reply:
Are you sure you want to use _that_ ~~IDE~~ editor?.. :D

Reply:
I like that aprroach makes sense

Reply:
I recently learned that you do not have to declare such dumb classes just to mock a function. An alternative is to declare some members right in the mock declaration.

```dart
class Function1Mock<T0, T1> extends Mock {
  T0 call(T0 arg);
}
```

At least we have such declarations in our internal codebase. Maybe `=> null` body can work, didn't try.